tests/test_interface.py

```diff
@@ -461,3 +461,762 @@ def test_InterfacesCreateRemove(self, dvs, testlog):
+
+        self.verify_appdb_ip_intf_addr_absence("Ethernet12")
+
+        self.verify_asicdb_ip_intf_addr_absence("fc00:1:1::/64", "fc00:1:1::1/128", False)
+
+
+    #
+    # IPv6 link-local partial-overlap scenario:
+    #
+    # intf #1: fe80:1:1::1/64
+    # intf #2: fe80:1:1::2/64
+    #
+    # Success criteria:
+    #
+    # - Only one IPsubnet route ("fe80:1:1::/64") must be injected in asicDB
+    #   corresponding to the first interface being created.
+    # - Both IP2me routes ("fe80:1:1::1/128" & "fe80:1:1::2/128") must be injected
+    #   into asicDB corresponding to the two interfaces being created.
+    # - Verify internal router-intfs are created/eliminated.
+    #
+    def test_IPv6LinkLocalInterfacePartialOverlap(self, dvs, testlog):
+
+        self.setup_db(dvs)
+
+        # Add a regular ipv4 address to interface #1
+        self.add_ip_address("Ethernet8", "fe80:1:1::1/64", "IPv6", "local")
+
+        self.verify_appdb_ip_intf_addr_presence("Ethernet8", "fe80:1:1::1/64", "IPv6", "local")
+
+        self.verify_asicdb_ip_intf_addr_presence("fe80:1:1::/64", "fe80:1:1::1/128", 1)
+
+        #
+        # Adding overlap ip-address to a interface #2
+        #
+        self.add_ip_address("Ethernet12", "fe80:1:1::2/64", "IPv6", "local")
+
+        self.verify_appdb_ip_intf_addr_presence("Ethernet12", "fe80:1:1::2/64", "IPv6", "local")
+
+        self.verify_asicdb_ip_intf_addr_presence("fe80:1:1::/64", "fe80:1:1::2/128", 2)
+
+        #
+        # Remove overlapped ip-address from interface #2
+        #
+        self.remove_ip_address("Ethernet12", "fe80:1:1::2/64")
+
+        self.verify_appdb_ip_intf_addr_absence("Ethernet12")
+
+        self.verify_asicdb_ip_intf_addr_absence("fe80:1:1::/64", "fe80:1:1::2/128", True)
+
+        #
+        # Remove ip-address from original interface #1
+        #
+        self.remove_ip_address("Ethernet8", "fe80:1:1::1/64")
+
+        self.verify_appdb_ip_intf_addr_absence("Ethernet8")
+
+        self.verify_asicdb_ip_intf_addr_absence("fe80:1:1::/64", "fe80:1:1::1/128", False)
+
+
+    #
+    # IPv6 link-local partial-overlap scenario (reversed):
+    #
+    # intf #1: fe80:1:1::1/64
+    # intf #2: fe80:1:1::2/64
+    #
+    # Success criteria:
+    #
+    # - Same as above. The difference this time is that, as we are eliminating the
+    #   interface with the primary IPsubnet route ("fe80:1:1::/64"), our implementation
+    #   must 'resurrect' the equivalent subnet-route associated to the secondary
+    #   intf.
+    #
+    def test_IPv6LinkLocalInterfacePartialOverlapReversed(self, dvs, testlog):
+
+        self.setup_db(dvs)
+
+        # Add a regular ipv4 address to interface #1
+        self.add_ip_address("Ethernet8", "fe80:1:1::1/64", "IPv6", "local")
+
+        self.verify_appdb_ip_intf_addr_presence("Ethernet8", "fe80:1:1::1/64", "IPv6", "local")
+
+        self.verify_asicdb_ip_intf_addr_presence("fe80:1:1::/64", "fe80:1:1::1/128", 1)
```

Comment:
The prefixes you are using for IPv6 link-local are illegal prefixes. Can you please correct?